components/automate-ui/src/app/pages/policy/add-members/policy-add-members.component.ts

```diff
@@ -305,4 +332,74 @@ export class PolicyAddMembersComponent implements OnInit, OnDestroy {
     this.addOrRemoveQueuedMember(true, member);
     this.closeModal();
   }
+
+  public showInputs(inputName: string): boolean {
+    const typeValue = this.expressionForm.get('type').value;
+    const identityValue = this.expressionForm.get('identity').value;
+
+    switch (inputName) {
+      case 'identity':
+        if (typeValue === 'USER' || typeValue === 'TEAM') {
+            this.allIdentity = typeValue;
+            this.updateValidations('identity', true);
+            return true;
+          } else {
+            this.updateValidations('identity', false);
+            this.expressionForm.get('identity').reset();
+            return false;
+          }
+        break;
+      case 'name':
+        if ( typeValue === 'TOKEN' || (identityValue && identityValue !== '*') ) {
+          this.updateValidations('name', true);
+          return true;
+        } else {
+          this.updateValidations('name', false);
+          this.expressionForm.get('name').reset();
+          return false;
+        }
+        break;
+      default:
+        return false;
+    }
+  }
+
+  private updateValidations(inputName: string, active: boolean): void {
+    const updateInput = this.expressionForm.get(inputName);
+
+    active === true
```

Comment:
Nothing invalid with using the ternary this way but convention in the code base is to use it only when you are using its return value; otherwise use a regular `if-then-else`. Then it is also more clear when you just say `if (active)...` without the `=== true`.